File: generate_images.js
#!/usr/bin/env node
/**
 * Generate PNG images for TutoFacile tutorials
 * Creates colored gradient images for each tutorial
 * Uses canvas library or pure binary PNG generation
 */

const fs = require('fs');
const path = require('path');

// PNG file signature
const PNG_SIGNATURE = Buffer.from([137, 80, 78, 71, 13, 10, 26, 10]);

/**
 * Create a simple PNG file with gradient
 * Returns a Buffer containing the PNG data
 */
function createGradientPNG(width, height, startColor, endColor) {
  // This is complex - let's use a simpler approach with SVG->PNG conversion
  // For now, we'll create SVG files and document how to convert them
  
  // Create SVG content
  const svg = `<svg xmlns='http://www.w3.org/2000/svg' width='${width}' height='${height}'>
    <defs>
      <linearGradient id='grad' x1='0%' y1='0%' x2='100%' y2='100%'>
        <stop offset='0%' style='stop-color:rgb(${startColor[0]},${startColor[1]},${startColor[2]});stop-opacity:1' />
        <stop offset='100%' style='stop-color:rgb(${endColor[0]},${endColor[1]},${endColor[2]});stop-opacity:1' />
      </linearGradient>
    </defs>
    <rect width='${width}' height='${height}' fill='url(#grad)'/>
  </svg>`;
  
  return svg;
}

// Tutorial configurations
const tutorials = [
  {
    id: 1,
    filename: 'tuto-1-flexbox',
    startColor: [0, 105, 246],
    endColor: [0, 196, 167],
    text: 'CSS',
    fontSize: 60
  },
  {
    id: 2,
    filename: 'tuto-2-cookies',
    startColor: [34, 197, 94],
    endColor: [245, 158, 11],
    text: '🍪',
    fontSize: 60
  },
  {
    id: 3,
    filename: 'tuto-3-nodejs',
    startColor: [0, 105, 246],
    endColor: [94, 45, 253],
    text: 'N',
    fontSize: 60
  },
  {
    id: 4,
    filename: 'tuto-4-tomates',
    startColor: [239, 68, 68],
    endColor: [245, 158, 11],
    text: '🍅',
    fontSize: 60
  },
  {
    id: 5,
    filename: 'tuto-5-chocolate',
    startColor: [139, 69, 19],
    endColor: [210, 105, 30],
    text: '🍫',
    fontSize: 60
  }
];

const imagesDir = path.join(__dirname, 'images');

// Create images directory
if (!fs.existsSync(imagesDir)) {
  fs.mkdirSync(imagesDir, { recursive: true });
  console.log('✅ Created images directory');
}

// Generate SVG files (these can be converted to PNG with ImageMagick or online tools)
console.log('Generating SVG files...\n');

tutorials.forEach(tutorial => {
  try {
    const svg = `<svg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 400 300'>
  <defs>
    <linearGradient id='grad' x1='0%' y1='0%' x2='100%' y2='100%'>
      <stop offset='0%' style='stop-color:rgb(${tutorial.startColor[0]},${tutorial.startColor[1]},${tutorial.startColor[2]});stop-opacity:1' />
      <stop offset='100%' style='stop-color:rgb(${tutorial.endColor[0]},${tutorial.endColor[1]},${tutorial.endColor[2]});stop-opacity:1' />
    </linearGradient>
  </defs>
  <rect width='400' height='300' fill='url(#grad)'/>
  <text x='200' y='150' font-size='${tutorial.fontSize}' fill='white' text-anchor='middle' dominant-baseline='middle' font-weight='bold'>${tutorial.text}</text>
</svg>`;
    
    const svgPath = path.join(imagesDir, `${tutorial.filename}.svg`);
    fs.writeFileSync(svgPath, svg, 'utf8');
    console.log(`✅ ${tutorial.filename}.svg`);
    
  } catch (err) {
    console.error(`❌ Error creating ${tutorial.filename}:`, err.message);
  }
});

console.log('\n📝 SVG files created successfully!');
console.log('\n💡 To convert SVG to PNG, you have several options:\n');
console.log('Option 1: Use an online converter');
console.log('   https://cloudconvert.com/svg-to-png\n');
console.log('Option 2: Use ImageMagick (command line)');
console.log('   convert tuto-1-flexbox.svg tuto-1-flexbox.png\n');
console.log('Option 3: Use Inkscape');
console.log('   inkscape tuto-1-flexbox.svg --export-png=tuto-1-flexbox.png\n');
console.log('Option 4: Use ffmpeg');
console.log('   ffmpeg -i tuto-1-flexbox.svg tuto-1-flexbox.png\n');

File: generate_images.py
#!/usr/bin/env python3
"""
Generate PNG images for TutoFacile tutorials
Creates colored gradient images for each tutorial
"""

from PIL import Image, ImageDraw, ImageFont
import os

# Define output directory
output_dir = os.path.dirname(os.path.abspath(__file__))
images_dir = os.path.join(output_dir, 'images')

# Create images directory if it doesn't exist
os.makedirs(images_dir, exist_ok=True)

# Tutorial image configurations
tutorials = [
    {
        'id': 1,
        'filename': 'tuto-1-flexbox.png',
        'gradient_start': (0, 105, 246),      # Blue #0069f6
        'gradient_end': (0, 196, 167),        # Teal #00c4a7
        'text': 'CSS',
        'emoji': None
    },
    {
        'id': 2,
        'filename': 'tuto-2-cookies.png',
        'gradient_start': (34, 197, 94),      # Green #22c55e
        'gradient_end': (245, 158, 11),       # Amber #f59e0b
        'text': '🍪',
        'emoji': True
    },
    {
        'id': 3,
        'filename': 'tuto-3-nodejs.png',
        'gradient_start': (0, 105, 246),      # Blue #0069f6
        'gradient_end': (94, 45, 253),        # Purple #5e2dfd
        'text': 'N',
        'emoji': None
    },
    {
        'id': 4,
        'filename': 'tuto-4-tomates.png',
        'gradient_start': (239, 68, 68),      # Red #ef4444
        'gradient_end': (245, 158, 11),       # Amber #f59e0b
        'text': '🍅',
        'emoji': True
    },
    {
        'id': 5,
        'filename': 'tuto-5-chocolate.png',
        'gradient_start': (139, 69, 19),      # Brown #8b4513
        'gradient_end': (210, 105, 30),       # Chocolate #d2691e
        'text': '🍫',
        'emoji': True
    },
]

def create_gradient_image(width, height, start_color, end_color):
    """Create an image with a linear gradient from top-left to bottom-right"""
    img = Image.new('RGB', (width, height))
    pixels = img.load()
    
    for y in range(height):
        for x in range(width):
            # Calculate gradient position (0.0 to 1.0)
            ratio = (x + y) / (width + height)
            
            # Interpolate color
            r = int(start_color[0] + (end_color[0] - start_color[0]) * ratio)
            g = int(start_color[1] + (end_color[1] - start_color[1]) * ratio)
            b = int(start_color[2] + (end_color[2] - start_color[2]) * ratio)
            
            pixels[x, y] = (r, g, b)
    
    return img

def generate_images():
    """Generate all tutorial images"""
    for tutorial in tutorials:
        print(f"Generating {tutorial['filename']}...", end=' ')
        
        # Create gradient background
        img = create_gradient_image(400, 300, tutorial['gradient_start'], tutorial['gradient_end'])
        draw = ImageDraw.Draw(img)
        
        # Add text/emoji in center
        text = tutorial['text']
        font_size = 80 if tutorial['emoji'] else 60
        
        try:
            # Try to use a default system font
            if tutorial['emoji']:
                # For emoji, use larger default font
                font = ImageFont.load_default()
            else:
                font = ImageFont.load_default()
        except:
            font = ImageFont.load_default()
        
        # Get text bounding box for centering
        bbox = draw.textbbox((0, 0), text, font=font)
        text_width = bbox[2] - bbox[0]
        text_height = bbox[3] - bbox[1]
        
        x = (400 - text_width) // 2
        y = (300 - text_height) // 2
        
        # Draw text in white
        draw.text((x, y), text, fill=(255, 255, 255), font=font)
        
        # Save image
        filepath = os.path.join(images_dir, tutorial['filename'])
        img.save(filepath, 'PNG')
        
        print(f"✅ Saved to {filepath}")

if __name__ == '__main__':
    try:
        generate_images()
        print("\n✅ All images generated successfully!")
    except Exception as e:
        print(f"\n❌ Error generating images: {e}")
        import traceback
        traceback.print_exc()
